include/vrv/object.h

```diff
@@ -871,41 +939,47 @@ class Object : public BoundingBox {
     virtual int AdjustTempo(FunctorParams *) { return FUNCTOR_CONTINUE; }
 
     /**
-     * @name Adjust the x position of a right barline in order to make sure the is no text content
+     * Adjust the x position of a right barline in order to make sure the is no text content
      * overlflowing in the right margin
      */
-    ///@{
     virtual int AdjustXOverflow(FunctorParams *) { return FUNCTOR_CONTINUE; }
+
+    /**
+     * End Functor for Object::AdjustXOverflow
+     */
     virtual int AdjustXOverflowEnd(FunctorParams *) { return FUNCTOR_CONTINUE; }
-    ///@}
 
     /**
      * Lay out the X positions of the staff content looking at the bounding boxes.
      * The functor process by aligned-staff content, that is from a rediction in the
      * MeasureAligner and then staff by staff but taking into account cross-staff elements
      */
-    ///@{
     virtual int AdjustXPos(FunctorParams *) { return FUNCTOR_CONTINUE; }
+
+    /**
+     * End Functor for Object::AdjustXPos
+     */
     virtual int AdjustXPosEnd(FunctorParams *) { return FUNCTOR_CONTINUE; }
-    ///@}
 
     /**
      * Adjust the spacing of the syl processing verse by verse
      */
-    ///@{
     virtual int AdjustSylSpacing(FunctorParams *) { return FUNCTOR_CONTINUE; };
+
+    /**
+     * End Functor for Object::AdjustSylSpacing
+     */
     virtual int AdjustSylSpacingEnd(FunctorParams *) { return FUNCTOR_CONTINUE; };
-    ///@}
 
     /**
      * Calculate the x position of tuplet brackets and num
      */
     virtual int AdjustTupletsX(FunctorParams *) { return FUNCTOR_CONTINUE; }
 
     /**
-     * Calculate the slur direction
+     * Cache or restore cached horizontal layout for faster layout redoing
      */
-    virtual int CalcSlurDirection(FunctorParams *) { return FUNCTOR_CONTINUE; }
+    virtual int LayOutHorizontallyCache(FunctorParams *) { return FUNCTOR_CONTINUE; }
```

Comment:
"Horizontally" is an adverb, but it's modifying a noun here; Perhaps either `LayOutHorizontalCache` or `LayOutHorizontallyCached`

If this is only for caching, perhaps it would be clearer to move `Cache` to the front?

`CacheHorizontalLayOut`?

Reply:
Yes, I was scratching my head for this one. This is what we call in `Doc::CastOffDocBase`
https://github.com/rism-digital/verovio/blob/6e5213b585ef63278a5da9c13b0c1de483b45d5e/src/doc.cpp#L911-L921

`Page::LayOutHorizontallyCachePage` is used for caching but also restoring the cache and calls this functor. I would suggest to:
* Rename `Page::LayOutHorizontallyCachePage` to Page::LayOutHorizontallyWithCache`
* Rename the functor `LayOutHorizontallyCache` to `CacheHorizontalLayout`

Does it look any clearer?

Reply:
Yes, that looks better!

Reply:
"Layout" and "Lay out" -- Although English usage isn't 100% consistent, the former is a noun, and the latter is a verb ("to lay out").

Reply:
Yes, the idea was to have a verb